python/taichi/lang/ops.py

```diff
@@ -369,30 +376,50 @@ def round(x):  # pylint: disable=redefined-builtin
         >>>     print(ti.round(x))
         [-2., 1., 3.]
     """
-    return _unary_operation(_ti_core.expr_round, builtins.round, x)
+    result = _round(x)
+    if dtype is not None:
+        result = cast(result, dtype)
+    return result
 
 
 @unary
-def floor(x):
+def _floor(x):
+    return _unary_operation(_ti_core.expr_floor, math.floor, x)
+
+
+def floor(x, dtype=None):
     """Return the floor of the input, element-wise.
     The floor of the scalar `x` is the largest integer `k`, such that `k <= x`.
+
     Args:
         x (Union[:mod:`~taichi.types.primitive_types`, :class:`~taichi.Matrix`]): \
             Input scalar or matrix.
+
+        dtype: (:mod:`~taichi.types.primitive_types`): the returned type, default to `None`. If \
+            set to `None` the retuned value will have the same type with `x`.
+
     Returns:
-        The floor of each element in `x`, with float type.
+        The floor of each element in `x`, with specified data type.
```

Comment:
```suggestion
        The floor of each element in `x`, with return value type `dtype`.
```
Please modify other places too.